packages/overlays/src/OverlayController.js

```diff
@@ -0,0 +1,192 @@
+import { fakeExtendsEventTarget } from './utils/fake-extends-target.js';
+import { GlobalOverlayController } from './GlobalOverlayController.js';
+import { LocalOverlayController } from './LocalOverlayController.js';
+
+// const loadGlobalOverlayController = async() => import('./GlobalOverlayController.js');
+// const loadLocalOverlayController = async() => import('./LocalOverlayController.js');
+
+/**
+ * @see https://popper.js.org/
+ * @typedef {object} PopperConfig
+ * @property {'auto'|'top'|'right'|'bottom'|'left'|'top-start'|'top-end'|'right-start'|'right-end'|'bottom-start'|'bottom-end'|'left-start'|'left-end'} [placement='bottom']
+ * @property {boolean} [positionFixed=false]
+ * @property {boolean} [eventsEnabled=true]
+ * @property {boolean} [removeOnDestroy=false]
+ * @property {object} [modifiers]
+ * @property {Function} [onCreate]
+ * @property {Function} [onUpdate]
+ */
+
+export class OverlayController {
+  /**
+   * @typedef {object} OverlayConfig
+   * @property {HTMLElement} [elementToFocusAfterHide=document.body] - the element that should be
+   * called `.focus()` on after dialog closes
+   * @property {boolean} [hasBackdrop=false] - whether it should have a backdrop (currently
+   * exclusive to globalOverlayController)
+   * @property {boolean} [isBlocking=false] - hides other overlays when mutiple are opened
+   * (currently exclusive to globalOverlayController)
+   * @property {boolean} [preventsScroll=false] - prevents scrolling body content when overlay
+   * opened (currently exclusive to globalOverlayController)
+   * @property {boolean} [trapsKeyboardFocus=false] - rotates tab, implicitly set when 'isModal'
+   * @property {boolean} [hidesOnEsc=false] - hides the overlay when pressing [ esc ]
+   * @property {boolean} [hidesOnOutsideClick=false] - hides the overlay when clicking next to it,
+   * exluding invoker. (currently exclusive to localOverlayController)
+   * https://github.com/ing-bank/lion/pull/61
+   * @property {TemplateResult} [contentTemplate]
+   * @property {TemplateResult} [invokerTemplate] (currently exclusive to LocalOverlayController)
+   * @property {HTMLElement} invokerNode
+   * @property {HTMLElement} contentNode
+   * @property {boolean} [isModal=false] - sets aria-modal and/or aria-hidden="true" on siblings
+   * @property {boolean} [isGlobal=false] - determines the connection point in DOM (body vs next
+   * to invoker). This is what other libraries often refer to as 'portal'. TODO: rename to renderToBody?
+   * @property {boolean} [isTooltip=false] - has a totally different interaction- and accessibility pattern from all other overlays, so needed for internals.
```

Comment:
hmmm doesn't `isTooltip` "forces" it to be a local overlay?

Reply:
`isToolTip` is a very confusing flag?
this seems like promoting a single mega class that has all the features and we enable different parts via options... 

not so sure if that is a smart idea?

Reply:
We already use it like this in the LocalOverlayController. I agree that that one (already) feels heavy for a tooltip, but the reason it is needed is that accessibility is just way different than in other overlays. Ergo: if we want a solution that is accessible out of the box, the local overlay **should** know we are dealing with `[role=tooltip]` 
If local means 'attached as sibling of invoker' then I think indeed that this one forces that (that is when we want to use aria-describedby and aria-labelledby). This a11y constraint doesn't hold for dropdowns (just like Popper.js allows you to render your dropdowns to the body)

But indeed, some flags 'trigger' other flags. If we deliver the right presets this should be very straightforward for the developer. There are several loose ends in the total Overlay system still, I think;  **for simplicity, we can leave the isTooltip out now.** 
Maybe we can have a separate discussion about improvements below?

### Improvements controllers

- Could be more balanced (especially local vs global) 
    - We totally position local overlays (even deliver ‘pointing frames’), but we don’t deliver any positioning for global overlays **(already being worked on atm)**
    - We only partly deliver accessibility (we should either do 100% or not at all)
    - We don’t distinguish between tooltips and others (a11y)
    - We only partly do user interaction. If we want to handle accessibility out of the box, we  should be fully aware of all interactions and not leave clicks and focuses up to the user.
- Viewport paddings are not customisable on constructor level, but do heavily depend on Design Systems
- It should be possible to deliver own ‘portals’, default one is not customisable and highly opinionated (backdrop colour, opacity, animation etc.)
- The default portal doesn’t use shadow DOM and has potential style leaks
- Support for sr-only as hiding mechanism: think of a navigation menu at the top of your page that should be fully indexed by screen readers 

### Improvements components (normalization of apis for Subclassers)

Universal interface mixing for elements wrapping overlay system (select, autocomplete, tooltip, contextmenu, popover)
- Create predictability and consistency for App Developers:
    - opened state pointing to isShown
    - Forward public api ’showOverlay’, ‘hideOverlay’, ’toggleOverlay’
- Create predictability for Subclassers:
    - _overlay holds overlay controller. A subclasser can always rely on this interface, regardless whether he has an overlay relative to viewport or the invoker and regardless of whether the content is placed in a portal or not.
    - _defineOverlay In this method the overlay controller is returned. This can be a dropdown for a select that transforms into a bottom sheet on mobile. The overlay controller allows to hook into events(like ‘before-show’) to change to the desired overlay type. When a portal switch takes place, the overlay controller takes care of keeping focus in sync.

Reply:
But it comes down a bit to the question I think: do we want our system to handle accesibility or not :)
Currently it does, but not in all cases. This feels a bit arbitrary to me. Think we should either do it fully (aligns with our mission to make solutions that are accessible out of the box) or we shouldn't do it at all.
For me, extracting all different 'accessibility scenarios' based on w3c specs beforehand is what brings the extra added value in this system. (there is just a limited set of accessible widgets mappable to our overlays and by baling them into the system we create predictable user experiences for impaired users).

We can also say that we extract all accessibility in a different layer? Like an OverlayA11yMixin?
Then you can still always use the barebone layer when user interaction might not be involved.